src/quote/indexQuoteProvider.ts

```diff
@@ -122,6 +131,48 @@ export class FlashMintQuoteProvider
           tx,
         }
       }
+      case FlashMintContractType.leveragedExtended: {
+        if (!zeroExApiV1) {
+          throw new Error('Contract type requires ZeroExApiV1 to be defined')
+        }
+        const leverageExtendedQuoteProvider =
+          new LeveragedExtendedQuoteProvider(provider, zeroExApiV1)
+        const leveragedExtendedQuote =
+          await leverageExtendedQuoteProvider.getQuote(request)
+        if (!leveragedExtendedQuote) return null
+        const builder = new LeveragedExtendedTransactionBuilder(provider)
+        const txRequest: FlashMintLeveragedExtendedBuildRequest = {
+          isMinting,
+          inputToken: inputToken.address,
+          inputTokenSymbol: inputToken.symbol,
+          outputToken: outputToken.address,
+          outputTokenSymbol: outputToken.symbol,
+          inputTokenAmount: BigNumber.from(0), // TODO: leveragedExtendedQuote.inputTokenAmount,
+          outputTokenAmount: BigNumber.from(0), // TODO: leveragedExtendedQuote.outputTokenAmount,
+          swapDataDebtCollateral: leveragedExtendedQuote.swapDataDebtCollateral,
+          swapDataInputOutputToken: leveragedExtendedQuote.swapDataPaymentToken,
+          swapDataInputTokenForETH:
+            leveragedExtendedQuote.swapDataDebtCollateral, // FIXME: add swapDataInputTokenForETH (returned from quote provider)
+          // FIXME: check if new arguments should be set here in the SDK or rather passed as arguments to they can be set from the app
+          priceEstimateInflator: wei(0.9),
+          maxDust: wei(0.00001),
+        }
+        const tx = await builder.build(txRequest)
+        if (!tx) return null
+        return {
+          chainId,
+          contractType,
+          /* eslint-disable  @typescript-eslint/no-non-null-assertion */
+          contract: tx.to!,
+          isMinting,
+          inputToken,
+          outputToken,
+          indexTokenAmount,
+          inputOutputAmount: leveragedExtendedQuote.inputOutputTokenAmount, // TODO: check
```

Comment:
Check that `indexTokenAmount` and `inputOutputAmount` are also the correct values here that we return as quotes to the app.

When done all the above it would be good to create a test in `indexQuoteProvider.test.ts` for ETH2x. I usually don't do this for every token because this will be indirectly tested with the integrations tests in `/tests` (another TODO of the PR).